src/stdune/id.mli

```diff
@@ -0,0 +1,44 @@
+open !Set
+
+type t
+
+(** Convert an ID to an integer. *)
+val to_int : t -> int
+
+(** The internal IdGen module which generates an
+    id using an instance of an ID generator for
+    the next ID. *)
+module IdGen : sig
```

Comment:
I don't think this should be exposed, it's not used anywhere.

Reply:
This isn't used, but I thought it might be useful if some object would like none local ids, e.g. if internal rules would like to generate local ids for dependencies it could do something like:

```
type t = {
  idgen : IdGen.t
}

...
let add_dep rule =
  ... let id = IdGen.gen rule.idgen in ...
```

Otherwise I can simplify it as you mentioned below, getting rid of all the id gen things.

Reply:
Yh, let's simplify it. I prefer to keep dune lean for now.